catalog/rocks/staging/iterator.go

```diff
@@ -0,0 +1,68 @@
+package staging
+
+import (
+	"context"
+
+	"github.com/treeverse/lakefs/catalog/rocks"
+)
+
+const batchSize = 1000
+
+type Iterator struct {
+	mgr         *Manager
+	nextFrom    rocks.Path
+	idxInBuffer int
+	err         error
+	ctx         context.Context
+	st          rocks.StagingToken
+	dbHasNext   bool
+	buffer      []*rocks.EntryRecord
+}
+
+func NewIterator(ctx context.Context, mgr *Manager, st rocks.StagingToken, nextFrom rocks.Path) *Iterator {
+	return &Iterator{nextFrom: nextFrom, mgr: mgr, ctx: ctx, st: st, dbHasNext: true}
+}
+
+func (s *Iterator) Next() bool {
+	s.idxInBuffer++
+	if s.idxInBuffer >= len(s.buffer) {
+		if !s.dbHasNext {
+			return false
+		}
+		res, err := s.mgr.listEntries(s.ctx, s.st, s.nextFrom, batchSize)
+		if err != nil {
+			s.err = err
+			return false
+		}
+		if len(res.entries) == 0 {
```

Comment:
do we need to set has next to false there are no results so we will not request again list entries on empty result?